templates/macros.html

```diff
@@ -22,8 +22,29 @@
 {# Makes the appropriate CSS imports for highlighting #}
 {% macro highlight_css() %}
     {# Load the highlighting theme css #}
-    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.4.0/styles/github.min.css"
-        type="text/css" media="all" />
+    <script>
+        // Choose which highlight.js theme to load based on the user theme
+        var stylesheet;
+        switch(document.documentElement.dataset.theme) {
+            case "dark":
+            case "ayu":
+                stylesheet = "https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.4.0/styles/dark.min.css";
+                break;
+            case "light":
```

Comment:
A `switch`/`case` in javascript 😰 

I think this either needs `theme.js` to set the theme to `light` explicitly when there is none, or include `null` in this case (and I have no idea if `case null` works).

Reply:
The `default` was intentional, in case someone adds a theme without updating this switch it will give an error and they'll have to update it. I'm ok with defaulting to 'light' but `case null` doesn't seem right.

Reply:
This isn't for unknown cases, it's for when you initially load docs.rs before rustdoc has stored a theme to use. If you check in a private browser it doesn't set it to `light`.

Actually, it sets it to `data-theme="null"`, so `case "null":` would work 🤣

Reply:
Oh! Good catch, thanks :) I hadn't thought about private mode.